### PR TITLE
Test1 pull request a class for the database.

### PR DESCRIPTION
**Pull request by ben188ben188**
pull request a class for the database.

### CODE REVIEW

File: portfolio/pom.xml
Comment (perrycate) [inline portfolio/pom.xml]: This line is going to cause a merge conflict with [Matt's PR](https://github.com/Mmendero/SPS-Team-Project-Name-TBD-/pull/1) - remember, you guys are working in the same repository, this is all shared code.

File: portfolio/src/main/java/com/google/sps/data/Information.java
Comment (perrycate) [inline portfolio/src/main/java/com/google/sps/data/Information.java]: Delete this commented out line.

File: portfolio/src/main/java/com/google/sps/data/Information.java
Comment (perrycate) [inline portfolio/src/main/java/com/google/sps/data/Information.java]: Is there a more descriptive class name we could come up with? "Information" is really generic, it could mean anything. Matt and Matthew may have some ideas if you're having trouble thinking of one.

File: portfolio/src/main/java/com/google/sps/data/Information.java
Comment (perrycate) [inline portfolio/src/main/java/com/google/sps/data/Information.java]: Can you come up with a comment describing what this class does/what information it holds.

File: portfolio/src/main/java/com/google/sps/data/Information.java
Comment (perrycate) [inline portfolio/src/main/java/com/google/sps/data/Information.java]: If this constructor isn't being used, you can probably just delete it.

File: portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java
Comment (perrycate) [inline portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java]: Why "/inf"?

File: portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java
Comment (perrycate) [inline portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java]: You can remove this commented out line.

File: portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java
Comment (perrycate) [inline portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java]: These aren't comments anymore, right? If so then you'll want to update this.

File: portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java
Comment (perrycate) [inline portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java]: What is this method doing? If you want to convert to json, why not use the "Gson().toJson()" stuff from around line 50?

File: portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java
Comment (perrycate) [inline portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java]: Since technically all of the parameters in your request are input, you could probably leave out the "-input" from the parameter names.

File: portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java
Comment (perrycate) [inline portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java]: Typo: I think you mean "import.html", not "inport.html"

File: portfolio/src/main/webapp/script.js
Comment (perrycate) [inline portfolio/src/main/webapp/script.js]: What is this function for? It seems like it's not being used yet.

File: portfolio/src/main/webapp/inport.html
Comment (perrycate) [inline portfolio/src/main/webapp/inport.html]: Same note as above - this file should probably be called "import.html", not "inport.html".

File: portfolio/src/main/webapp/inport.html
Comment (perrycate) [inline portfolio/src/main/webapp/inport.html]: Since Matt is working on the html stuff, it may be worth checking with him to make sure that this file is needed.

File: portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java
Comment (perrycate) [inline portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java]: Pardon the potentially silly question: I'm not that familiar with this DataStore thing. Is there anything we need to change somewhere to add the new properties (like type), or will it just work automatically?

**Review by perrycate — COMMENTED** [review]
There are some small things to tidy up before merging, but overall this looks good. Good work!

**Comment by perrycate** [comment]
Woohoo here are those merge conflicts I mentioned a few weeks ago. Learning opportunity incoming. Let me know if you need help resolving these.

File: portfolio/src/main/java/com/google/sps/data/Information.java
Comment (matthewtlam) [reply portfolio/src/main/java/com/google/sps/data/Information.java]: Maybe a better class name would be PostInformation or RecommendationInfo?

File: portfolio/src/main/java/com/google/sps/data/Information.java
Comment (matthewtlam) [inline portfolio/src/main/java/com/google/sps/data/Information.java]: Possibly something we could do is use the builder format for our class since there are so many parameters in our constructor. I can try to help with that later on

File: portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java
Comment (matthewtlam) [reply portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java]: I think we should use something more descriptive than the short form for information. Possibly take the suggestions from my class names that I suggested

File: portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java
Comment (matthewtlam) [inline portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java]: Possibly comment to explain what this block of code is doing

File: portfolio/src/main/webapp/script.js
Comment (matthewtlam) [reply portfolio/src/main/webapp/script.js]: I think the function is for taking data from the servlet and pushing it to the front end

File: portfolio/src/main/webapp/inport.html
Comment (Mmendero) [reply portfolio/src/main/webapp/inport.html]: I was planning on creating a separate page for account management/creating more listings of items/resources. This seems fine considering this is on a separate .html file. However I am unsure on as to what the "getHistory()" functions does on load

File: portfolio/pom.xml
Comment (Mmendero) [reply portfolio/pom.xml]: I'm pretty sure there was a way to pass on or load variables from another file that you mentioned @perrycate , if so we could all just replace our unique ids with a similarly named placeholder variable could we not?

File: portfolio/src/main/java/com/google/sps/data/Information.java
Comment (perrycate) [reply portfolio/src/main/java/com/google/sps/data/Information.java]: Honestly I would say "Post" or "Recommendation" would work too. "Info" doesn't really add much, since technically everything we work with is information.

**Review by Mmendero — COMMENTED** [review]
It could be best to address the comments under the HomeServlet.java and Information.java before we continue with this pull request.

File: portfolio/pom.xml
Comment (ben188ben188) [reply portfolio/pom.xml]: Thank you guys， I will change it.

File: portfolio/src/main/java/com/google/sps/data/Information.java
Comment (ben188ben188) [reply portfolio/src/main/java/com/google/sps/data/Information.java]: OK, NP.

File: portfolio/src/main/java/com/google/sps/data/Information.java
Comment (ben188ben188) [reply portfolio/src/main/java/com/google/sps/data/Information.java]: Let me change it, How about PostInformation, I think this name from Matthew is OK.

File: portfolio/src/main/java/com/google/sps/data/Information.java
Comment (ben188ben188) [reply portfolio/src/main/java/com/google/sps/data/Information.java]: OK, I got it.

File: portfolio/src/main/java/com/google/sps/servlets/GetInformation.java
Comment (perrycate) [inline portfolio/src/main/java/com/google/sps/servlets/GetInformation.java]: typo: comments, not commnets

Comment (perrycate) [reply portfolio/src/main/java/com/google/sps/servlets/GetInformation.java]: Actually, wait a second. These aren't comments that it's storing anymore, are they? If so then this should probably  be updated with a more accurate description.

File: portfolio/src/main/java/com/google/sps/servlets/GetInformation.java
Comment (perrycate) [inline portfolio/src/main/java/com/google/sps/servlets/GetInformation.java]: Since there's no code here, I don't think you need to override it.

File: portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java
Comment (perrycate) [inline portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java]: Wait, I don't see a class named PostInformation, only GetInformation. Am I missing something? Is this a class that came with the starter code that I didn't notice?

**Review by perrycate — CHANGES_REQUESTED** [review]
Good progress, but I think there is still some previous feedback that needs to be addressed. Let me know if you run into any issues or are unsure about what needs to be done!

File: Site_Files/pom.xml
Comment (perrycate) [inline Site_Files/pom.xml:41]: You can delete these extra lines.

File: Site_Files/src/main/java/com/google/sps/servlets/GetInformation.java
Comment (perrycate) [inline Site_Files/src/main/java/com/google/sps/servlets/GetInformation.java:51]: I feel like I've asked this before, but I don't remember the answer: What is this function for?

File: Site_Files/src/main/java/com/google/sps/servlets/GetInformation.java
Comment (perrycate) [inline Site_Files/src/main/java/com/google/sps/servlets/GetInformation.java:62]: Since this is empty, you can remove it.

File: Site_Files/src/main/java/com/google/sps/servlets/HomeServlet.java
Comment (perrycate) [inline Site_Files/src/main/java/com/google/sps/servlets/HomeServlet.java:47]: Same question here as in the other servlet.

File: Site_Files/src/main/webapp/index.html
Comment (perrycate) [inline Site_Files/src/main/webapp/index.html:45]: Typo: "import"

File: Site_Files/src/main/webapp/script.js
Comment (perrycate) [inline Site_Files/src/main/webapp/script.js:31]: Is this still used even though the things are no longer comments? If so, consider renaming to something else.

File: Site_Files/src/main/webapp/script.js
Comment (perrycate) [inline Site_Files/src/main/webapp/script.js:63]: Nit: You can delete the extra lines.

**Review by perrycate — APPROVED** [review]
There's still a number of previous comments in additions to the ones I've left here that need to be addressed, but overall this looks good to go!

File: Site_Files/src/main/java/com/google/sps/servlets/GetInformation.java
Comment (Mmendero) [inline Site_Files/src/main/java/com/google/sps/servlets/GetInformation.java:62]: I'm a little confused. What is this function's main purpose?

**Review by Mmendero — APPROVED** [review]
I like the changes you've made to the website! Upon reading your code, however, I took a little bit to understand the intention/why you wrote certain lines of code. It could help to add comments describing the logic/algorithm of posting and getting information to and from the backend like in the GetInformation.java and HomeServlet.java. Aside from this everything looks good!

**Comment by perrycate** [comment]
For future reference: As a general principle I like to leave it to the original author of the commit to merge stuff back in, so they have time to address the feedback in the reviewer's comments before doing so.

**Pull request merged**
